### ISSUE TITLE
Projectiles do less damage if you also charged shot before they hit

### ISSUE THREAD

**Issue by strotlog** [dashrando/dash-rando-app#1]
There may be a tiny bug in the dash.ips damage code:

```asm
90/F63A:        ADA609          lda $09A6
90/F63D:        890010          bit #$1000
90/F640:        D011            bne $F653
90/F642:        BD2C0C          lda $0C2C,X
90/F645:        5E2C0C          lsr $0C2C,X
90/F648:        5E2C0C          lsr $0C2C,X
90/F64B:        6D2C0C          adc $0C2C
90/F64E:        4A              lsr A
90/F64F:        4A              lsr A
90/F650:        8D2C0C          sta $0C2C
90/F653:        BD180C          lda $0C18,X
90/F656:        60              rts
```

The `adc` and `sta` instructions should operate on `$0C2C,X` and not a constant `$0C2C`, which is simply the first projectile on the screen.

Usually this is fine, as it's hard to keep any other projectile on the screen for a full second - when you fire, your charged shot is that first and only projectile on the screen so everything works fine. If it's possible to release a charge with some other shot onscreen:
- The released charged shot would do only 25% of its standard damage, instead of 31.25%; and
- The original projectile on screen also gets significantly reduced damage. (It inherits 1/16 the standard charged shot's damage and combines this with 25% of its own original damage)

So a shooting missile + charged power beam simultaneously would do 28 + 15 damage, respectively, where they should do 100 + 18.

**Comment by strotlog** [comment]
See a related oddity in the the 33s clip linked below. It shows a shot and then a charged shot sharing the screen, first in vanilla and then in DASH (before collecting full charge beam). Damage values are displayed in decimal during each shot's travel.

https://gfycat.com/bowedfixedgrouse

I may have misinterpreted the code somewhat, as the damage values in memory watcher aren't behaving the way I predicted above. However, we do see the charged power beam can be 15 damage, which is 25% of standard damage, instead of the correct calculation (18 damage).

**Comment by dashrando** [comment]
Good find strotlog! Excellent break down of the issue too! This has been resolved in a recent commit (a97dcd7605c16c7e5658969b7cab415b1b45d5e6).
